python/pyspark/cloudpickle.py

```diff
@@ -318,6 +329,24 @@ def save_function(self, obj, name=None):
         Determines what kind of function obj is (e.g. lambda, defined at
         interactive prompt, etc) and handles the pickling appropriately.
         """
+        try:
+            should_special_case = obj in _BUILTIN_TYPE_CONSTRUCTORS
+        except TypeError:
+            # Methods of builtin types aren't hashable in python 2.
+            should_special_case = False
```

Comment:
BACKPORTED
BUG: Handle instance methods of builtin types https://github.com/cloudpipe/cloudpickle/pull/145